Add Quarkus documentation (#6138)
open-telemetry/opentelemetry.io: content/en/docs/zero-code/java/quarkus/getting-started.md

---
title: Getting started with Quarkus
weight: 20
---

To enable OpenTelemetry in your Quarkus application, add the
`quarkus-opentelemetry` extension dependency to your project.

{{< tabpane >}} {{< tab header="Maven (`pom.xml`)" lang=Maven >}}

> ````xml
> 
> {{< tabpane text=true >}} {{% tab header="Maven (`pom.xml`)" lang=Maven %}}
> 

Reviewer: This second and nested `tabpane ` seems to have been added by mistake (and it caused an extra escaping of the xml):

```suggestion
```

```xml
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-opentelemetry</artifactId>
</dependency>

> ````

Reviewer: ~~~suggestion
```
~~~

Author: Strange... I did run `npm run fix:all` on this [commit](https://github.com/open-telemetry/opentelemetry.io/pull/6138/commits/459409f3121a4e741ca8dc04a0dd0632206301d7) and it seems that it created these errors.

{{% /tab %}} {{% tab header="Gradle (`build.gradle`)" lang=Gradle %}}

```kotlin
implementation("io.quarkus:quarkus-opentelemetry")
```

{{% /tab %}} {{< /tabpane>}}

Only the **tracing** signal will be enabled by default. To enable **metrics**
and **logs**, you need to add the following configuration to your
`application.properties` file:

```properties
quarkus.otel.metrics.enabled=true
quarkus.otel.logs.enabled=true
```

OpenTelemetry logging is available after Quarkus 3.16.0.

The remaining configurations are available in the
[Quarkus OpenTelemetry configuration reference](https://quarkus.io/guides/opentelemetry#configuration-reference).

{{% alert title="Note" color="info" %}}

You can also use the [Java agent](../../agent) to instrument your Quarkus

> application in JVM mode. For the pros and cons, see

Reviewer 2: ```suggestion
application with a JVM. For the pros and cons, see
```

> [Java zero-code instrumentation](..).

Reviewer 2: There is nothing about it on the `Java zero-code instrumentation` part? We have written things about it for Spring. Perhaps we could remove this part for Quarkus today, and in another step see how we could provide pros and cons both for Spring and Quarkus.

Author: Removed that last paragraph and added a Quarkus mention to the Java zero-code instrumentation page.
